scheduler/src/cook/mesos/util.clj

```diff
@@ -198,6 +198,39 @@
      get-pending-jobs-duration
      (get-pending-job-ents* unfiltered-db true))))
 
+(defn generate-intervals
+  "Generates a list of intervals between start and end as pairs
+   [interval-start interval-end].
+
+   Parameters:
+   ----------
+   start : clj-time/datetime
+   end : clj-time/datetime
+   interval->steps : fn
+      function that takes a clj-time/interval and returns the number of steps
+      based on the step-fn between start and end
+   step-fn : fn
+      function that returns a clj-time/timedelta given n steps
+
+   Returns:
+   --------
+   list of pairs, [interval-start interval-end]"
+  ([start end]
+   (generate-intervals start end t/in-days t/days))
+  ([start end interval->steps step-fn]
+   (if (t/before? end start)
+     []
+     (->> (for [steps-from-start (range 0 (inc (interval->steps (t/interval start end))))]
```

Comment:
A `loop` feels a more natural choice to me here as we can quickly build off the next interval from the `end` of the last interval.